### DISCUSSION TITLE
Kafka topic not ready after increasing the number of partitions

### DISCUSSION THREAD

**Discussion by themkarimi** [strimzi/strimzi-kafka-operator#6218] [Q&A] [answered]
Hi,

I have increased the number of partitions of my kafka topic but after applying the new configuration, the topic status is stated " No Ready" instead an autogenerated topic with the pervious configuration is created and is working. my kafka cluster consists of 5 nodes and 3 zookeepers. does anyone have idea about this? 
the resources that I am talking about is kafkatopics.kafka.strimzi.io. 
[kafka-tp.log](https://github.com/strimzi/strimzi-kafka-operator/files/7888313/kafka-tp.log)

**Comment by scholzj** [comment]
I think you will need to share more information. The Topic operator logs ideally on the debug level, the resources you are talking about etc.

**Reply by themkarimi** [reply]
Thanks for your reply. I have added the a snippet of logs since they are too many stuff there.  I get alot of these messages for the topics that are faulty:
KafkaTopic is incompatible with the topic metadata. The topic metadata will be treated as canonical.

**Reply by scholzj** [reply]
@tombentley Any ideas?

**Reply by themkarimi** [reply]
![kafka](https://user-images.githubusercontent.com/89538066/150075209-26139c1d-087b-4b2c-be7f-e25c73149048.PNG)
for more information, this is the state of my topics!

**Comment by themkarimi** [comment, marked as answer]
The problem was fixed by adding generation: 2 to the metadata portion of my topic kubernetes manifest.  please update your documentation on examples/topic.

**Reply by scholzj** [reply]
The generation is managed by Kubernetes when you edit the custom resource. It might be just a coincidence that it helped in your case.

**Reply by themkarimi** [reply]
that's weird! maybe it is not totally fixed I  need to investigate more, at least those logs are not repeated anymore!
